tests/main/cloud-init/task.yaml

```diff
@@ -0,0 +1,45 @@
+summary: Ensure that cloud-init integration works
+
+description: |
+    snapd picks up basic cloud information from the host and makes it available
+    to the snaps. Run the test on a live backend which sets instance data
+    properly.
+
+# GCE backend sets instance data
+backends: [google]
+
+execute: |
+    if [[ ! -e /run/cloud-init/instance-data.json ]]; then
+        echo "cloud-init instance data is required to execute the test"
+        exit 0
```

Comment:
at least on the classic ubuntu systems this should exit 1, otherwise something might change under our feet and we would never notice

Reply:
This is resolved now.